src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -178,57 +180,26 @@ public PerIndexWriter newPerIndexWriter(StorageAttachedIndex index,
                                             RowMapping rowMapping,
                                             long keyCount)
     {
+        ComponentGroup.Writer groupWriter = indexDescriptor.newPerIndexGroupWriter(index.getIndexContext());
         // If we're not flushing or we haven't yet started the initialization build, flush from SSTable contents.
         if (tracker.opType() != OperationType.FLUSH || !index.canFlushFromMemtableIndex())
         {
             NamedMemoryLimiter limiter = SEGMENT_BUILD_MEMORY_LIMITER;
             logger.debug(index.getIndexContext().logMessage("Starting a compaction index build. Global segment memory usage: {}"),
                          prettyPrintMemory(limiter.currentBytesUsed()));
 
-            return new SSTableIndexWriter(indexDescriptor, index.getIndexContext(), limiter, index.isIndexValid(), keyCount);
+            return new SSTableIndexWriter(groupWriter, limiter, index.isIndexValid(), keyCount);
         }
 
         return new MemtableIndexWriter(index.getIndexContext().getPendingMemtableIndex(tracker),
-                                       indexDescriptor,
-                                       index.getIndexContext(),
+                                       groupWriter,
                                        rowMapping);
     }
 
-    @Override
-    public boolean validatePerSSTableComponents(IndexDescriptor indexDescriptor, boolean checksum)
-    {
-        for (IndexComponent indexComponent : perSSTableComponents())
-        {
-            if (isBuildCompletionMarker(indexComponent))
-                continue;
-
-            try (IndexInput input = indexDescriptor.openPerSSTableInput(indexComponent))
-            {
-                Version earliest = getExpectedEarliestVersion(indexComponent);
```

Comment:
The original code seems confusing. `perSSTableComponents()` won't return vector components. There is no point using `getExpectedEarliestVersion` here...

Reply:
That was true in the original (though I don't think it was hurting, and maybe the intent was that `getExpectedEarlierVersion` could later handle more cases and so calling in all case avoided future surprises).

In the update code though, the `validateIndexComponent` method is called for both per-sstable component and per-group components, so we need the call in general (we could, of course, check `component.group().isPerIndexGroup()` if we really wanted to only call `getExpectedEarlierVersion` on per-index groups, but as mentioned above, I don't think it really hurts to call it regardless).